backend/local/local.go

```diff
@@ -1573,32 +1573,46 @@ func (o *Object) SetMetadata(ctx context.Context, metadata fs.Metadata) error {
 }
 
 func cleanRootPath(s string, noUNC bool, enc encoder.MultiEncoder) string {
-	if runtime.GOOS != "windows" || !strings.HasPrefix(s, "\\") {
-		if !filepath.IsAbs(s) {
-			s2, err := filepath.Abs(s)
-			if err == nil {
-				s = s2
-			}
-		} else {
-			s = filepath.Clean(s)
-		}
-	}
+	var vol string
 	if runtime.GOOS == "windows" {
-		s = filepath.ToSlash(s)
-		vol := filepath.VolumeName(s)
+		vol = filepath.VolumeName(s)
 		if vol == `\\?` && len(s) >= 6 {
 			// `\\?\C:`
 			vol = s[:6]
 		}
-		s = vol + enc.FromStandardPath(s[len(vol):])
-		s = filepath.FromSlash(s)
-		if !noUNC {
-			// Convert to UNC
-			s = file.UNCPath(s)
+		s = s[len(vol):]
+	}
+	// Don't use FromStandardPath. Make sure Dot (`.`, `..`) as name will not be reencoded
+	if enc != encoder.Standard {
```

Comment:
Perhaps just mask out the `Dot` encoding from enc? That would make the code a lot simpler

something like this?

```
enc &= ^encoding.EncodeDot
```

If you can minimise this change I'm much more likely to  merge it!

Reply:
Consider someone want to create `./．` (Raw) in the current directory, the Standard encoding is `．/‛．`.
Then consider local-encoding with EncodeDot.
So given Standard: `./．/‛．` (the first dot means current directory), if use `enc &= ^encoding.EncodeDot`, it will be:
Standard: `./．/‛．` decoded to Raw: `././．` encoded to enc: `././．`, then Abs() lead to `．`.
But what we want write to the local filesystem is Raw: `．/‛．`.

I've added a comment to avoid anyone accidentally optimizing it by mistake.